packages/bentocache/src/cache/stack/cache_stack.ts

```diff
@@ -46,22 +48,23 @@ export class CacheStack {
     if (bus) return bus
     if (!busDriver || !this.l1) return
 
-    const opts = lodash.merge({ retryQueue: { enabled: true, maxSize: undefined } }, busOptions)
-    const newBus = new Bus(busDriver, this.l1, this.logger, this.emitter, opts)
+    this.#busDriver = busDriver
+    this.#busOptions = lodash.merge(
+      { retryQueue: { enabled: true, maxSize: undefined } },
+      busOptions,
+    )
+    const newBus = new Bus(this.#busDriver, this.l1, this.logger, this.emitter, this.#busOptions)
 
     return newBus
   }
 
   namespace(namespace: string) {
-    return new CacheStack(
-      this.name,
-      this.options,
-      {
-        l1Driver: this.l1?.namespace(namespace),
-        l2Driver: this.l2?.namespace(namespace),
-      },
-      this.bus,
-    )
+    return new CacheStack(this.name, this.options, {
+      l1Driver: this.l1?.namespace(namespace),
+      l2Driver: this.l2?.namespace(namespace),
+      busDriver: this.#busDriver,
+      busOptions: this.#busOptions,
+    })
```

Comment:
@gkachru  I believe, there should be 4th argument passed with `this.bus`, otherwise a new bus is created every time any namespace is accessed.

This can be seen in https://github.com/wodCZ/bentocache-bus-repro, if you run the `main.js`, you see the following error after 10 logs:
```
(node:70661) MaxListenersExceededWarning: Possible EventEmitter memory leak detected. 11 reconnecting listeners added to [Commander]. MaxListeners is 10. Use emitter.setMaxListeners() to increase limit
    at genericNodeError (node:internal/errors:983:15)
    at wrappedFn (node:internal/errors:537:14)
    at _addListener (node:events:593:17)
    at EventEmitter.addListener (node:events:611:10)
    at RedisTransport.onReconnect (file:///Users/wod/sites/bentotest/node_modules/@boringnode/bus/build/src/transports/redis.js:68:26)
    at new Bus (file:///Users/wod/sites/bentotest/node_modules/@boringnode/bus/build/src/bus.js:28:38)
    at new Bus (file:///Users/wod/sites/bentotest/node_modules/bentocache/build/index.js:588:17)
    at #createBus (file:///Users/wod/sites/bentotest/node_modules/bentocache/build/index.js:1098:20)
    at new _CacheStack (file:///Users/wod/sites/bentotest/node_modules/bentocache/build/index.js:1076:31)
    at _CacheStack.namespace (file:///Users/wod/sites/bentotest/node_modules/bentocache/build/index.js:1102:12)
```

Thanks for the amazing work otherwise 🙏

Reply:
PS: just checked adding the 4th arg locally and the memory leak warning is gone, while both namespace sync issues remain fixed.

Reply:
The bus does indeed need to be created every time, as you need to pass in the new namespace localcache instance. It is however generally a memory intensive way as both CacheStack and bus get created.

The event emitter warning can be eliminated by setting up something like `const bcProducts = bento.namespace('products')` and then using `bcProducts.set` etc from then on. However, still you would have as many CacheStack and bus instances as there are namespaces, plus one for the original bento instance.

The reason why `.clear()` still works as you pass in the original bus, is because all buses are listening to the same queue name! So the `clear` message is received by every instance of the bus, both the namespaced and non-namespaced one, which will clear all items everywhere!

Have noticed this, and I am working on this issue as well.

Reply:
Added PR #39 to address the excessive messages by namespacing the bus channel.